### ISSUE TITLE
[KIEKER-1234] Create hello world test for manual release test of Data Bridge

### ISSUE THREAD

**Issue by rju** [kieker-monitoring/kieker#2561]
JIRA Issue: [KIEKER-1234] Create hello world test for manual release test of Data Bridge
Original Reporter: Andre van Hoorn
 *** 
<p>Currently, the data bridge (script) is not tested as part of the release tests in the devguide.</p>

[KIEKER-1234]: https://kieker-monitoring.atlassian.net/browse/KIEKER-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by rju** [comment]
author André van Hoorn -- Thu, 30 Oct 2014 15:32:59 +0100
<p>An option is to call the HTTP-REST-based data bridge server via the databridge script (
    <span class="jira-issue-macro resolved" data-jira-key="KIEKER-1066" >
                <a href="https://kieker-monitoring.atlassian.net/browse/KIEKER-1066" class="jira-issue-macro-key issue-link"  title="HTTP PUT/REST API for data bridge" >
            <img class="icon" src="https://kieker-monitoring.atlassian.net/images/icons/issuetypes/genericissue.png" />
            KIEKER-1066
        </a>
                                                    <span class="aui-lozenge aui-lozenge-subtle aui-lozenge-success jira-macro-single-issue-export-pdf">Done</span>
            </span>
) and send a post request including JSON. </p>

**Comment by rju** [comment]
author André van Hoorn -- Fri, 4 Sep 2015 02:24:38 +0200
<p>Reiner Jung: Will you manage to to this for 1.12? Doesn't need to be sophisticated. If possible, just a couple of command according to the basic tests in wiki:create-release#QualityAssuranceandReleaseProcess</p>

**Comment by rju** [comment]
author rju -- Fri, 18 Sep 2015 21:07:53 +0200
<p>My general idea for this is to use a data blob of pre-recorded monitoring data and push it into the data bridge with <tt>netcat</tt>. To realize that the following must be build:</p>
<ul>
	<li>A pre recorded monitoring data blob for the bridge (I guess this can be achieved from the C client application once and stored in the repository)</li>
	<li>A start script which starts the data bridge in TCP single-server-mode and/or multi-server-mode
	<ul>
		<li>Start server</li>
		<li>Wait with netcat until the server is up</li>
		<li>Start the client</li>
		<li>Compare the resulting kieker log data with a pre computed sample<br/>
I'll try to build this weekend. Probably the start script could be expressed as two manual operations.<br/>
What do you think, is that a valid solution?</li>
	</ul>
	</li>
</ul>

**Comment by rju** [comment]
author rju -- Sat, 19 Sep 2015 17:27:48 +0200
<p>Added a limited test example, which tests if the application runs with tcp-server and tcp-single-server. I did not add tcp-client due to the extra effort in testing that. And actually we are testing the rest of the application already in the integration tests. Therefore, it would be pointless to test all connectors again.</p>

<p>The documentation can be found at: <a href="https://kieker-monitoring.atlassian.net/wiki/create-release/wrapper" class="external-link" rel="nofollow noreferrer">https://kieker-monitoring.atlassian.net/wiki/create-release/wrapper</a></p>

<p>Due to the excessive length of the segment which made it hard to understand anything and made it even harder to add and change text in future, I moved this test part to a separate page (link to the main page of QA has been added accordingly). </p>
